tests/e2e/RelatedModels.js

```diff
@@ -0,0 +1,45 @@
+import { Selector } from 'testcafe';
+import LayoutPage from './page-objects/Layout';
+import HomePage from './page-objects/Homepage';
+import RelatedModels from './page-objects/RelatedModels';
+
+const layoutPage = new LayoutPage();
+const homePage = new HomePage();
+const relatedModels = new RelatedModels();
+
+fixture('Loading related models while avoiding n+1 queries').page(
+  layoutPage.URL.staging
+);
+
+test('There are no liquid errors on the page', async t => {
+  await t.click(homePage.link.models);
+  await layoutPage.checkLiquidErrors();
+  await t.expect(relatedModels.link.documentation.exists).ok();
+});
+
+test('Loading related models while avoiding n+1 queries. Increase speed 10x', async t => {
+  await t
+    .click(homePage.link.models)
+    .expect(relatedModels.table.companies.count)
+    .eql(20)
+    .expect(relatedModels.table.benchmarkingResults.count)
+    .eql(3);
+  await t
+    .click(relatedModels.link.onlyProgrammers)
+    .expect(relatedModels.table.companies.count)
+    .eql(20)
+    .expect(relatedModels.table.benchmarkingResults.count)
+    .eql(3);
+  await t
+    .click(relatedModels.link.programmersCompaniesSlow)
+    .expect(relatedModels.table.companies.count)
+    .eql(20)
+    .expect(relatedModels.table.benchmarkingResults.count)
+    .eql(3);
+  await t
+    .click(relatedModels.link.programmersCompaniesCorrect)
+    .expect(relatedModels.table.companies.count)
+    .eql(20)
+    .expect(relatedModels.table.benchmarkingResults.count)
+    .eql(3);
+});
```

Comment:
Hmm. this page needs  to be tested differently

The whole point of those pages is to show that one solution is faster than the other and benchmark at the end of the page shows how much faster it is.

So comparing time from slow solution to time on fast solution is the way to test if it actually works (content in the table doesnt really matter as it might change and still prove the point - and i will probably change it when we migrate to new stack to show even bigger difference).

I will readd "total" row and make this table longer to make it more obvious how much faster it is.

Reply:
PS. For some reason the DB is empty on production, so this would fail :)

Reply:
Ok, so:

Compare total from: https://nearme-example.staging-oregon.near-me.com/programmers/with_companies_1

To total from: https://nearme-example.staging-oregon.near-me.com/programmers/with_companies_2

First should always be bigger than the second. You can even make it more interesting, by checking if second is at least 3x faster than the first one, as we are advertising it as being 10x faster.

Reply:
Thank you for explaining. However I noticed that the increase of speed is not entirely 10 times.
At the moment I used assertion "Less than" what should be ok in this case.

Reply:
Yeah, its not 10x, because the more data there is, the bigger the difference. In my tests it was 4-5x which is okay with dataset of 30*2